draft-ietf-tls-tls13.md

```diff
@@ -5450,6 +5505,32 @@ ClientHello without 0-RTT data.
 To avoid this error condition, multi-server deployments SHOULD ensure a uniform
 and stable deployment of TLS 1.3 without 0-RTT prior to enabling 0-RTT.
 
+## Middlebox Compatibility Mode {#middlebox}
+
+Field measurements have found that a significant number of middleboxes
+misbehave when a TLS client/server pair negotiates TLS 1.3. Implementations
+can increase the chance of making connections through those middleboxes
+by making the TLS 1.3 handshake look more like a TLS 1.2 handshake:
+
+- The client always provides a non-empty session ID in the ClientHello.
+
+- If not offering early data, the client sends a dummy
+  change_cipher_spec record immediately before its second flight. This
+  may either be before its second ClientHello or before its encrypted
```

Comment:
A somewhat irksome consequence of this: if you do completely stateless HRR, you need to skip the CCS before parsing your "first" (actually second) ClientHello and picking the version.

(Then again, we only need this over TCP which is stateful...)